packages/api/src/qm/visitor.ts

```diff
@@ -2,135 +2,76 @@
 // Licensed under the MIT license.
 
 import {
-  Func,
   TextInputQuestion,
-  NodeType,
   QTreeNode,
   Question,
   SingleSelectQuestion,
-  Option,
-  StaticOption,
+  StaticOptions,
   OptionItem,
-  MultiSelectQuestion,
-  FileQuestion,
-  DynamicValue,
-  AnswerValue,
+  MultiSelectQuestion
 } from "./question";
-import { getValidationFunction, RemoteFuncExecutor, validate } from "./validation";
-import { ConfigMap } from "../config";
-import { InputResult, UserInteraction } from "./ui";
+import { getValidationFunction, validate } from "./validation";
 import { returnSystemError, returnUserError } from "../error";
-
-async function getRealValue(
-  parentValue: unknown,
-  defaultValue: unknown,
-  inputs: ConfigMap,
-  remoteFuncExecutor?: RemoteFuncExecutor
-): Promise<unknown> {
-  let output: unknown = defaultValue;
-  if (typeof defaultValue === "string") {
-    const defstr = defaultValue as string;
-    if (defstr === "$parent") {
-      output = parentValue;
-    } else if (defstr.startsWith("$parent.") && parentValue instanceof Object) {
-      const property = defstr.substr(8);
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      output = (parentValue as any)[property];
-    }
-  } else {
-    output = await getCallFuncValue(inputs, false, defaultValue as Func, remoteFuncExecutor);
-  }
-  return output;
-}
+import { Inputs } from "../types";
+import { InputResult, UserInteraction } from "./ui";
 
 export function isAutoSkipSelect(q: Question): boolean {
-  if (q.type === NodeType.singleSelect || q.type === NodeType.multiSelect) {
-    const select = q as SingleSelectQuestion | MultiSelectQuestion;
-    const options = select.option as StaticOption;
-    if (select.skipSingleOption && select.option instanceof Array && options.length === 1) {
+  if (q.type === "singleSelect" || q.type === "multiSelect") {
```

Comment:
It's a good practice to use constants, rather than strings. SingleSelectQuestion.type is better than "singleSelect".